framework/contracts/account/src/modules.rs

```diff
@@ -347,30 +339,65 @@ fn _whitelist_modules(deps: DepsMut, modules: Vec<String>) -> AccountResult<()>
 }
 
 /// Remove a contract from the whitelist
-fn _remove_whitelist_module(deps: DepsMut, module: String) -> AccountResult<()> {
-    WHITELISTED_MODULES.update(deps.storage, |mut whitelisted_modules| {
-        let module_address = deps.api.addr_validate(&module)?;
-
-        if !whitelisted_modules.0.contains(&module_address) {
-            return Err(AccountError::NotWhitelisted(module.clone()));
-        }
-        // Remove contract from whitelist.
-        whitelisted_modules.0.retain(|addr| *addr != module_address);
-        Ok(whitelisted_modules)
-    })?;
-
+pub(crate) fn _remove_whitelist_modules(
+    deps: DepsMut,
+    module_addresses: Vec<Addr>,
+) -> AccountResult<()> {
+    for module_addr in module_addresses.into_iter() {
+        WHITELISTED_MODULES.update(deps.storage, |mut whitelisted_modules| {
+            if !whitelisted_modules.0.contains(&module_addr) {
+                return Err(AccountError::NotWhitelisted(module_addr.to_string()));
+            }
+            // Remove contract from whitelist.
+            whitelisted_modules.0.retain(|addr| *addr != module_addr);
+            Ok(whitelisted_modules)
+        })?;
+    }
     Ok(())
 }
```

Comment:
Let's have single load or move loop inside `update`. This will `load`+`save` `WHITELISTED_MODULES` for every module, instead of 1 load and 1 save

Reply:
Not pushed? 👀

Reply:
pushed :)

Reply:
> Better now? x)

Yeah, nice!